salt/modules/status.py

```diff
@@ -366,8 +366,12 @@ def nproc():
 
         salt '*' status.nproc
     '''
-    data = __salt__['cmd.run']('nproc')
+    cmd = {
+        'Linux': 'nproc',
+        'FreeBSD': 'sysctl -n hw.ncpu',
+    }
     try:
+        data = __salt__['cmd.run'](cmd[__grains__['kernel']])
```

Comment:
Actually, I am sorry, I should have thought of this sooner, we should change this to just return the grain, since the code is redundant, then this function is very easy:

```
return __grains__.get('nproc', 0)
```